pkg/kubectl/resource/BUILD

```diff
@@ -21,10 +21,10 @@ go_library(
         "//build/visible_to:pkg_kubectl_resource_CONSUMERS",
     ],
     deps = [
-        "//pkg/api:go_default_library",
```

Comment:
still has this as a test dependency below

Reply:
Yes.  I would like to get rid of that dependency, but I think this PR on its own frees us up by allowing projects to vendor this code without vendoring all of Kubernetes.  Here is what I am thinking:

- move this code to the kubernetes/common repo
- kubernetes/common vendors all of kubernetes/kubernetes for testing :(
- other repos vendor kubernetes/common, but not the transitive test dependencies

Reply:
agree it can be a follow up

Reply:
+1 on improving our test coverage and structure.  In general, being able to develop individual components and have confidence that they work correctly independently of testing other components is critical to project health and scaling.

Do you have any idea how complex this would be?  Is it something that would make sense to give to a new contributor?